client/src/app/core/core-services/model-request-builder.service.ts

```diff
@@ -210,57 +252,80 @@ export class ModelRequestBuilderService implements OnAfterAppsLoaded {
         }
         const isSpecificStructuredField = queryIdField !== effectiveIdField;
 
-        const relation: Relation | null = this.relationManager.findRelation(collection, queryIdField);
+        const relation: Relation | null = this.relationManager.findRelation(
+            modelRequestObject.collection,
+            queryIdField
+        );
         if (!relation) {
             throw new Error(
-                `Relation with ownIdField ${queryIdField} (effective ${effectiveIdField}) in collection ${collection} unknown!`
+                `Relation with ownIdField ${queryIdField} (effective ${effectiveIdField}) in collection ${modelRequestObject.collection} unknown!`
             );
         }
 
-        let descriptor: FieldDescriptor;
+        let response: DescriptorResponse<
+            RelationFieldDescriptor | GenericRelationFieldDecriptor | StructuredFieldDecriptor
+        >;
         if (!relation.generic && (!relation.structured || isSpecificStructuredField)) {
-            descriptor = this.getRelationFieldDescriptor(relation, follow);
+            response = this.getRelationFieldDescriptor(relation, follow);
         } else if (relation.generic) {
-            descriptor = this.getGenericRelationFieldDescriptor(relation, follow);
+            response = this.getGenericRelationFieldDescriptor(relation, follow);
         } else {
-            descriptor = this.getStructuredFieldDescriptor(relation, follow);
+            response = this.getStructuredFieldDescriptor(relation, follow);
         }
 
-        fields[effectiveIdField] = descriptor;
+        modelRequestObject.fields[effectiveIdField] = response.descriptor;
+        modelRequestObject.collectionsToUpdate.push(...response.collectionsToUpdate);
```

Comment:
Maybe a set as the type for `collectionsToUpdate` would be nice to avoid duplicates